addons/purchase_stock/models/stock_rule.py

```diff
@@ -118,11 +118,15 @@ def _run_buy(self, procurements):
 
     @api.model
     def _get_procurements_to_merge_groupby(self, procurement):
-        return procurement.product_id, procurement.product_uom, procurement.values['propagate_date'], procurement.values['propagate_date_minimum_delta'], procurement.values['propagate_cancel']
+        return procurement.product_id, procurement.product_uom, procurement.values['propagate_date'],\
+            procurement.values['propagate_date_minimum_delta'], procurement.values['propagate_cancel'],\
+            (procurement.values.get('orderpoint_id') and not procurement.values.get('move_dest_ids')) and procurement.values['orderpoint_id']
```

Comment:
why the conditions on move_dest_ids?

Reply:
you should only have one procurement max per op right? these conditons do not seem useful

Reply:
Do not merge procurements with an orderpoint_id if they don't have move_dest_ids. linked to commit 22b9343c9ac84099ab500b5617f14c1c3d1e4c05
Since they are all lauched as a batch we won't merge procurement from different order point without move_dest_ids. If they have a move_dest we don't care since 1. The destination location does not depends of the order line. 2 the quantity in progress do not use the order line but the internal moves.